nmigen_boards/arty_a7.py

```diff
@@ -0,0 +1,251 @@
+            Subsignal("scl_pullup", Pins("A14", dir="o")),
+            Subsignal("sda_pullup", Pins("A13", dir="o")),
+            Attrs(IOSTANDARD="LVCMOS33")
+        ),
+
+        *SPIFlashResources(0,
+            cs="L13", clk="L16", mosi="K17", miso="K18", wp="L14", hold="M14",
+            attrs=Attrs(IOSTANDARD="LVCMOS33")
+        ),
+
+        Resource("ddr3", 0,
+            Subsignal("rst",    PinsN("K6", dir="o")),
+            Subsignal("clk",    DiffPairs("U9", "V9", dir="o"), Attrs(IOSTANDARD="DIFF_SSTL135")),
+            Subsignal("clk_en", Pins("N5", dir="o")),
+            Subsignal("cs",     PinsN("U8", dir="o")),
+            Subsignal("we",     PinsN("P5", dir="o")),
+            Subsignal("ras",    PinsN("P3", dir="o")),
+            Subsignal("cas",    PinsN("M4", dir="o")),
+            Subsignal("a",      Pins("R2 M6 N4 T1 N6 R7 V6 U7 R8 V7 R6 U6 T6 T8", dir="o")),
+            Subsignal("ba",     Pins("R1 P4 P2", dir="o")),
+            Subsignal("dqs",    DiffPairs("N2 U2", "N1 V2", dir="io"),
+                                Attrs(IOSTANDARD="DIFF_SSTL135")),
+            Subsignal("dq",     Pins("K5 L3 K3 L6 M3 M1 L4 M2 V4 T5 U4 V5 V1 T3 U3 R3", dir="io"),
+                                Attrs(IN_TERM="UNTUNED_SPLIT_40")),
+            Subsignal("dm",     Pins("L1 U1", dir="o")),
+            Subsignal("odt",    Pins("R5", dir="o")),
+            Attrs(IOSTANDARD="SSTL135", SLEW="FAST"),
+        ),
+
+        Resource("eth_clk25", 0, Pins("G18", dir="o"),
+                 Clock(25e6), Attrs(IOSTANDARD="LVCMOS33")),
+        Resource("eth_clk50", 0, Pins("G18", dir="o"),
+                 Clock(50e6), Attrs(IOSTANDARD="LVCMOS33")),
+        Resource("eth_mii", 0,
+            Subsignal("rst",     PinsN("C16", dir="o")),
+            Subsignal("mdio",    Pins("K13", dir="io")),
+            Subsignal("mdc",     Pins("F16", dir="o")),
+            Subsignal("tx_clk",  Pins("H16", dir="i")),
+            Subsignal("tx_en",   Pins("H15", dir="o")),
+            Subsignal("tx_data", Pins("H14 J14 J13 H17", dir="o")),
+            Subsignal("rx_clk",  Pins("F15", dir="i")),
+            Subsignal("rx_dv",   Pins("G16", dir="i"), Attrs(PULLDOWN="TRUE")),
+            Subsignal("rx_er",   Pins("C17", dir="i")),
+            Subsignal("rx_data", Pins("D18 E17 E18 G17", dir="i")),
+            Subsignal("col",     Pins("D17", dir="i")),
+            Subsignal("crs",     Pins("G14", dir="i")),
+            Attrs(IOSTANDARD="LVCMOS33")
+        ),
+        Resource("eth_rmii", 0,
+            Subsignal("rst",       PinsN("C16", dir="o")),
+            Subsignal("mdio",      Pins("K13", dir="io")),
+            Subsignal("mdc",       Pins("F16", dir="o")),
+            Subsignal("tx_en",     Pins("H15", dir="o")),
+            Subsignal("tx_data",   Pins("H14 J14", dir="o")),
+            Subsignal("rx_crs_dv", Pins("G14", dir="i")),
+            Subsignal("rx_dv",     Pins("G16", dir="i"), Attrs(PULLUP="TRUE")),
```

Comment:
OK, I believe there is only one remaining review item. Could you explain what is `rx_dv` here? AFAIK, in RMII there is only `rx_crs_dv`.

Reply:
The PHY (TI DP83848J) can operate in RMII mode by strapping `rx_dv` high during reset.
`rx_dv` can also be used in RMII mode as a simpler way to access DV, without having to extract it from `rx_crs_dv`.

This is according to the datasheet, I actually haven't tested RMII on the Arty.